test/widgets/cards/summary_preview_card_test.dart

```diff
@@ -0,0 +1,132 @@
+        'Then it should display "Income" and primary background color',
+        (WidgetTester tester) async {
+          // Arrange
+          const amount = '5000';
+          const percentage = '10';
+
+          await tester.pumpWidget(createWidgetUnderTest(
+            header: HeaderType.income,
+            amount: amount,
+            percentage: percentage,
+          ));
+
+          expect(find.text('Income'), findsOneWidget);
+
+          final headerContainer = tester.widget<Container>(
+            find.descendant(
+                of: find.byType(SummaryPreviewCard),
+                matching: find.byType(Container).last),
+          );
+
+          expect(
+            (headerContainer.decoration as BoxDecoration).color,
+            appTheme.colors.primary,
+          );
+        },
+      );
+
+      testWidgets(
+        'Given SummaryPreviewCard with HeaderType.expense '
+        'Then it should display "Expense" and secondary background color',
+        (WidgetTester tester) async {
+          // Arrange
+          const amount = '2000';
+          const percentage = '5';
+
+          await tester.pumpWidget(createWidgetUnderTest(
+            header: HeaderType.expense,
+            amount: amount,
+            percentage: percentage,
+          ));
+
+          // Assert
+          expect(find.text('Expense'), findsOneWidget);
+          final container = tester.widget<Container>(find
+              .descendant(
+                  of: find.byType(SummaryPreviewCard),
+                  matching: find.byType(
+                    Container,
+                  ))
+              .last);
+
+          expect(
+            (container.decoration as BoxDecoration).color,
+            appTheme.colors.secondary,
+          );
+        },
+      );
+
+      testWidgets(
+        'Given SummaryPreviewCard '
+        'Then it should display the correct amount with ₹ symbol',
+        (WidgetTester tester) async {
+          // Arrange
+          const amount = '5000';
+          const percentage = '10';
+
+          await tester.pumpWidget(createWidgetUnderTest(
+            header: HeaderType.income,
+            amount: amount,
+            percentage: percentage,
+          ));
+
+          // Assert
+          expect(find.text('₹5000'), findsOneWidget);
+        },
+      );
+
+      testWidgets(
+        'Given SummaryPreviewCard '
+        'Then it should display the correct percentage &"vs last month" text',
+        (WidgetTester tester) async {
+          // Arrange
+          const amount = '5000';
+          const percentage = '10';
+
+          await tester.pumpWidget(createWidgetUnderTest(
+            header: HeaderType.income,
+            amount: amount,
+            percentage: percentage,
+          ));
+
+          final textWidgets =
+              tester.widgetList<Text>(find.byType(Text)).toList();
+
+          expect(textWidgets[1].data, contains('₹$amount'));
```

Comment:
₹5,000 should be expected not ₹5000.
also the test case is about rendering the vs. last month; so expect that text is rendered properly too